pkg/cluster/cluster.go

```diff
@@ -496,3 +504,20 @@ func IsUnroutable(host string) bool {
 	}
 	return false
 }
+
+func makeConfig(networkType string) *memberlist.Config {
+	var mc *memberlist.Config
+
+	switch networkType {
+	case DefaultLanPeerType:
+		mc = memberlist.DefaultLANConfig()
+	case DefaultWanPeerType:
+		mc = memberlist.DefaultWANConfig()
+	case DefaultLocalPeerType:
+		fallthrough
+	default:
+		mc = memberlist.DefaultLocalConfig()
```

Comment:
Let's stick to only one place of choosing a default. Flag seems better for this as you have now, but that means that we need to return error here. 

Otherwise we can have bug of not setting `networkType` somewhere and it will be hard to notice it.